### PR TITLE
Add Dev Container Configuration

### PR DESCRIPTION
**Pull request by aaron-lane**
## Summary

This pull request adds a dev container configuration which can be used [locally with VS Code](https://code.visualstudio.com/docs/devcontainers/containers) or [remotely with GitHub Codespaces](https://docs.github.com/en/codespaces/overview). The dev container is intended to include all of the dependencies required to develop the site.

### reboot/settings.py

- adding a hook for `CSRF_TRUSTED_ORIGINS`

### .env.sample

- adding `127.0.0.1` to `ALLOWED_HOSTS` for local development
- adding `CSRF_TRUSTED_ORIGINS=.preview.app.github.dev` for remote development
- adding `SECURE_SSL_REDIRECT=False` for local development

### Makefile

- adding `codespace`, `.env`, and `.git/hooks/pre-commit` targets to assist with building the dev container
- adding conditional usage of `sudo` with RabbitMQ commands to the `env` and `stopenv` targets for compatibility with the dev container

### make-venv

- switching from `sh` to `bash` for compatibility with the dev container

### app/migrations/0019_auto_20230523_2116.py

This migration file was automatically generated during the initial setup. Presumably, it was missed from being committed in a previous commit.

### CODE REVIEW

File: app/migrations/0019_auto_20230523_2116.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.2.28 on 2023-05-24 01:16
```
Comment (bwdmonkey) [inline app/migrations/0019_auto_20230523_2116.py:1]: hey @aaron-lane, thanks for the effort and introducing the containerization changes :) I think the rest of the changes LGTM, just slightly worried about this migration as it contains duplicate changes to https://github.com/CodeTheChangeUBC/reBOOT/blob/master/app/migrations/0017_auto_20200903_1754.py

Could we remove change for now and check why it's introducing this first before we add it?

**Review by bwdmonkey — APPROVED** [review]

File: app/migrations/0019_auto_20230523_2116.py (hunk 1 above)
Comment (aaron-lane) [reply app/migrations/0019_auto_20230523_2116.py]: I am not qualified at this point to speculate as to why this migration was automatically generated during development setup, but I noticed that it is subtly different, as it adds the `('CABLES', 'Cables')` and `('RECYCLE', 'Recycle')` tuples which are not present in the original migration.

For now, I've reverted the addition of the migration in https://github.com/CodeTheChangeUBC/reBOOT/pull/337/commits/64bdd0272ec56872d7d6738e55d1b35ca22557ad, as requested.

Comment (bwdmonkey) [reply app/migrations/0019_auto_20230523_2116.py]: Oh good catch! Feel free to squash merge when you are ready then :) we can add the migration later then

**Comment by aaron-lane** [comment]
@bwdmonkey I'm not authorized to merge. Would you like to do so, or grant me write access to the repository? ✍🏼

**Comment by bwdmonkey** [comment]
> @bwdmonkey I'm not authorized to merge. Would you like to do so, or grant me write access to the repository? ✍🏼

@aaron-lane gave access :) https://github.com/CodeTheChangeUBC/reBOOT/invitations

**Pull request merged**